lib/core/src/system/web/dom/html/css3d_renderer.rs

```diff
@@ -0,0 +1,357 @@
+//! This module contains the Css3dRenderer, a struct used to render CSS3D elements.
+
+use crate::prelude::*;
+
+use crate::display::object::DisplayObjectData;
+use crate::display::camera::Camera2d;
+use crate::display::camera::camera2d::Projection;
+use crate::system::web::dom::html::{Css3dObject, Css3dSystem};
+use crate::system::gpu::data::JsBufferView;
+use crate::system::web::Result;
+use crate::system::web::create_element;
+use crate::system::web::dyn_into;
+use crate::system::web::NodeInserter;
+use crate::system::web::NodeRemover;
+use crate::system::web::StyleSetter;
+use crate::system::web::dom::DomContainer;
+use crate::system::web::dom::ResizeCallback;
+use crate::system::web::get_element_by_id;
+use super::css3d_object::Css3dOrder;
+
+use nalgebra::Vector2;
+use nalgebra::Matrix4;
+use wasm_bindgen::prelude::wasm_bindgen;
+use wasm_bindgen::JsValue;
+use web_sys::HtmlElement;
+use js_sys::Object;
+
+
+// ===================
+// === Js Bindings ===
+// ===================
+
+mod js {
+    use super::*;
+    #[wasm_bindgen(inline_js = "
+        function arr_to_css_matrix3d(a) {
+            return 'matrix3d(' + a.join(',') + ')'
+        }
+
+        // Sets object's CSS 3D transform.
+        export function set_object_transform(dom, matrix_array) {
+            let css = arr_to_css_matrix3d(matrix_array);
+            dom.style.transform = 'translate(-50%, -50%)' + css;
+        }
+
+        // Setup perspective CSS 3D projection on DOM.
+        export function setup_perspective(dom, perspective) {
+            dom.style.perspective = perspective + 'px';
+        }
+
+        // Setup Camera orthographic projection on DOM.
+        export function setup_camera_orthographic(dom, matrix_array) {
+            dom.style.transform = arr_to_css_matrix3d(matrix_array);
+        }
+
+        // Setup Camera perspective projection on DOM.
+        export function setup_camera_perspective
+        (dom, near, matrix_array) {
+            let translateZ  = 'translateZ(' + near + 'px)';
+            let matrix3d    = arr_to_css_matrix3d(matrix_array);
+            let transform   = translateZ + matrix3d;
+            dom.style.transform = transform;
+        }
```

Comment:
I was writing about it in my review - can we move doc comments form JS to Rust? RThis way rust doc gen contains it :)